pkg/controller/account/iam.go

```diff
@@ -117,17 +117,79 @@ func getStsCredentials(reqLogger logr.Logger, client awsclient.Client, iamRoleNa
 	return assumeRoleOutput, nil
 }
 
+func retryIfAwsServiceFailureOrInvalidToken(err error) bool {
+	if aerr, ok := err.(awserr.Error); ok {
+		switch aerr.Code() {
+		// ServiceFailure may be an unspecified server-side error, and is worth retrying
+		case "ServiceFailure":
+			return true
+		// InvalidClientTokenId may be a transient auth issue, retry
+		case "InvalidClientTokenId":
+			return true
+		}
+	}
+	// Otherwise, do not retry
+	return false
+}
+
+func listAccessKeys(client awsclient.Client, iamUser *iam.User) (*iam.ListAccessKeysOutput, error) {
+	var result *iam.ListAccessKeysOutput
+	var err error
+
+	// Default is 1/10 of a second, but any retries we need to make should be delayed a few seconds
+	// This also defaults to an exponential backoff, so we only need to try ~5 times, default is 10
+	retry.DefaultDelay = 3 * time.Second
+	retry.DefaultAttempts = uint(5)
```

Comment:
Hm, I just noticed: these seem to be global? That's kind of icky. Not a big deal, probably, since we're setting the same wherever we're using them. But does the library provide a way to set them explicitly per use?

Reply:
From my quick once-over of the library: no, there's not a way to set them explicitly per use.  I might have missed it though :(

Reply:
I didn't dig in too far. It looks like it's certainly possible based on the examples, but it wouldn't be worthwhile if we had to write huge custom handlers like those.

Anyway, like I said, it's not a huge deal since we're setting them the same everywhere -- but to emphasize the fact that they're global, perhaps we should do those assignments (once) at the top level of the file?

Reply:
We can revisit the scope of those variables when we review the general retry approach for AWS